app/controllers/auth_source_ldaps_controller.rb

```diff
@@ -18,6 +18,22 @@ def create
     end
   end
 
+  def test_connection
+    result = {}
+    begin
+      temp_auth_source_ldap = AuthSourceLdap.new(params[:auth_source_ldap])
+      temp_auth_source_ldap.ldap_con.valid_user?("")
```

Comment:
@eLobato : actually the result that it returns is immaterial. `ldap_fluff` doesnt actually try creating a connection with just `AuthSourceLdap.new(params[:auth_source_ldap])` , it tries connection only after we try to perform an operation on the said ldap server, say check if a user is valid, hence I had added `temp_auth_source_ldap.ldap_con.valid_user?("")` to run a dummy query, and hence try and initiate a connection.